tests/README.rst

```diff
@@ -3,51 +3,32 @@
 Crate Docker Image Development
 ==============================
 
-Running Tests
+Prerequisites
 -------------
 
-Create a virtual environment::
-
-  >>> python3 -m venv .venv
-
-Then activate it::
-
-  >>> source .venv/bin/activate
-
-Now use pip to install the dependencies::
+For invoking the next steps, you will need to have Docker and Python installed
+on your computer.
 
-  >>> pip install -r requirements.txt
+First, create a virtual environment and install the Python dependencies::
 
-The tests require `Docker <https://www.docker.com>`_ version `1.10.x`
-with API version `1.22` to be installed on the local machine::
+    python3 -m venv .venv
+    source .venv/bin/activate
+    pip install -r requirements.txt
 
-  >>> docker version
-  Client:
-   Version:      1.10.3
-   API version:  1.22
-   Go version:   go1.5.3
-   Git commit:   20f81dd
-   Built:        Thu Mar 10 21:49:11 2016
-   OS/Arch:      darwin/amd64
-  Server:
-   Version:      1.10.3
-   API version:  1.22
-   Go version:   go1.5.3
-   Git commit:   20f81dd
-   Built:        Thu Mar 10 21:49:11 2016
-   OS/Arch:      linux/amd64
+Also, please check `adjust system limits on Linux`_, before proceeding with
+invoking the test suite::
 
-For OSX you would need to install `Docker Toolbox <https://www.docker.com/products/docker-toolbox>`_
-and initialize the Docker environment using the Quickstart Terminal::
+    sudo sysctl -w vm.max_map_count=262144
 
-  >>> /bin/bash -c "/Applications/Docker/Docker\ Quickstart\ Terminal.app/Contents/Resources/Scripts/start.sh"
 
-For CrateDB >= 1.2 you would need to change the following setting in your docker machine::
-
-  >>> docker-machine ssh default "sudo sysctl -w vm.max_map_count=262144"
```

Comment:
Just double checking, is this not required anymore (to set the limit inside docker machine?)

Reply:
There is still a prose section left about this detail, together with the corresponding command how to adjust that value on the host machine, `sudo sysctl -w vm.max_map_count=262144`.

On the other hand, `docker-machine` is a thing of the past, no? At least, I don't seem to have it on my macOS machine, so I removed that sentence here.

Reply:
I will be happy to improve this section. For example, without reading it thorougly, [1] has some guidelines specifically for Windows/WSL and ES.

[1] https://stackoverflow.com/questions/69214301/using-docker-desktop-for-windows-how-can-sysctl-parameters-be-configured-to-per

Reply:
Please also check this: https://www.anycodings.com/1questions/448018/update-maxmapcount-for-elasticsearch-docker-container-mac-host which refers to docker machine to set the limit.

Reply:
Dear Marios,

thank you for that pointer. I think the specific section where the `docker-machine` command is reflected within the _On Docker Toolbox_ section, is outdated information. [_Docker Toolbox_](https://github.com/docker-archive/toolbox) is surely a thing of the past:

> ⚠️ Deprecation Notice: This project and repository is now deprecated and is no longer in active development. Please use [Docker Desktop](https://www.docker.com/products/docker-desktop) instead where possible.

With kind regards,
Andreas.

Reply:
> I will be happy to improve this section.

In the _On Latest Docker For Mac_ section of the document you referenced, this snippet is mentioned:
```
screen ~/Library/Containers/com.docker.docker/Data/vms/0/tty
sysctl -w vm.max_map_count=262144
```

However, it did not work for me and croaked with `Permission denied`. I think good resources in this area are [^1] and [^2], where it looks like this would be the right incantation to access the Linux host machine within Docker Desktop in order to adjust those values these days.

```
docker run -it --privileged --pid=host debian nsenter -t 1 -m -u -n -i sh
sysctl -w vm.max_map_count=262144
```

[^1]: https://github.com/docker/for-mac/issues/4822#issuecomment-766762967
[^2]: https://dev.to/moyarich/how-to-find-the-storage-location-of-docker-images-and-containers-on-a-mac-369i

Reply:
By the way, `docker run --rm -it python:3.9 sysctl vm.max_map_count` already says `vm.max_map_count = 262144` on my machine, apparently by default now?

```
$ docker run -it --rm --privileged --pid=host justincormack/nsenter1
/ # uname -a
Linux docker-desktop 5.10.104-linuxkit #1 SMP Thu Mar 17 17:08:06 UTC 2022 x86_64 Linux
```

I can confirm the setting is not made manually by looking at the output of
```
docker run -it --rm --privileged --pid=host justincormack/nsenter1 /bin/sh -c "/bin/cat /etc/sysctl.d/*"
```

So, maybe the need to raise this system setting is also a thing of the past already?

Reply:
### Observation
I've been able to clarify the situation on Win10. On a default installation, it says:
```
PS C:\Windows\System32\WindowsPowerShell\v1.0> docker run --rm -it python:3.9 sysctl vm.max_map_count
vm.max_map_count = 65530
```

### Solution
In your Windows `%userprofile%` directory (typically `C:\Users\<username>`), create or edit the file `.wslconfig` with the following:

```ini
[wsl2]
kernelCommandLine = "sysctl.vm.max_map_count=262144"
```

Then exit any WSL instance, invoke `wsl --shutdown`, and restart.

```
> sysctl vm.max_map_count
vm.max_map_count = 262144
```

-- https://stackoverflow.com/a/69294687
-- https://docs.microsoft.com/en-us/windows/wsl/wsl-config#wslconfig

Reply:
Verified that for latest macs the `vm.max_map_count` is already `262144` by default.
My system:
<img width="576" alt="Screenshot 2022-09-13 at 16 37 38" src="https://user-images.githubusercontent.com/5058131/189918879-020f0f9a-7019-48e3-be3a-99b3a1640983.png">

Reply:
As promised, I'll double check the windows guide on my windows 11 and post here.

Reply:
Verified that: 
- for `Windows 11`  **with** `wsl2` the `max_map_count` by default is `65530` and the `.wslconfig`, as described in the update guide, sets it correctly
- for `Window 11` **without** `wsl2` the `max_map_count` by default is `262144` and no further configuration is required.

Reply:
Thanks, I've converged your findings into a less-detailed remark like
> Depending on the Docker backend you are using on Windows (WSL2 vs. Hyper-V),
the setting may show a default value of ``65530``, or ``262144``.
>
> In order to adjust the value, add this [....]